Source/NSWindow.m

```diff
@@ -2701,6 +2705,50 @@ - (NSScreen *) screen
   return _screen;
 }
 
+- (void) applicationDidChangeScreenParameters: (NSNotification *)aNotif
+{
+  NSRect oldScreenFrame = [_screen frame];
+  int    screenNumber = [_screen screenNumber];
+  NSRect newScreenFrame;
+  NSRect newFrame;
+
+  // NSLog(@"[NSWindow - %@] Old/New Screen RC: %lu/%lu",
+  //       [self className], [_screen retainCount],
+  //       [[[NSScreen screens] objectAtIndex: 0] retainCount]);
+  
+  // We need to get new screen from renewed screen list because
+  // [NSScreen mainScreen] returns NSScreen object of key window and that object
+  // will never be released.
+  for (NSScreen *scr in [NSScreen screens])
```

Comment:
Not sure whether this is supported by all compilers we want to work with.

Reply:
Changed to NSEnumerator.